File: test_gradio_client.mjs
import { Client } from "@gradio/client";

async function run() {
  try {
    const app = await Client.connect("levihsu/OOTDiffusion");
    
    const result = await app.predict("/process_hd", [
      "https://levihsu-ootdiffusion.hf.space/file=/tmp/gradio/2e0cca23e744c036b3905c4b6167371632942e1c/model_1.png", // filepath  in 'Model' Image component
      "https://levihsu-ootdiffusion.hf.space/file=/tmp/gradio/180d4e2a1139071a8685a5edee7ab24bcf1639f5/03244_00.jpg", // filepath  in 'Garment' Image component
      1, // number (numeric value between 1 and 4) in 'Images' Slider component
      20, // number (numeric value between 20 and 40) in 'Steps' Slider component
      2, // number (numeric value between 1.0 and 5.0) in 'Guidance scale' Slider component
      -1, // number (numeric value between -1 and 2147483647) in 'Seed' Slider component
    ]);
    
    console.log("SUCCESS:", JSON.stringify(result, null, 2));
  } catch (e) {
    console.error("ERROR:", e);
  }
}

run();

File: test_gradio_client.py
import requests

def download_image(url, filename):
    response = requests.get(url)
    if response.status_code == 200:
        with open(filename, 'wb') as f:
            f.write(response.content)

download_image("https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png", "human.png")
download_image("https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png", "garment.png")

from gradio_client import Client, handle_file

print("Initializing client...")
client = Client("levihsu/OOTDiffusion")
print("Client initialized. Predicting...")

try:
    result = client.predict(
        vton_img=handle_file('human.png'),
        garm_img=handle_file('garment.png'),
        n_samples=1,
        n_steps=20,
        image_scale=2.0,
        seed=-1,
        api_name="/process_hd"
    )
    print("Result HD:", result)
except Exception as e:
    print("Error in /process_hd:", e)

try:
    result_dc = client.predict(
        vton_img=handle_file('human.png'),
        garm_img=handle_file('garment.png'),
        category="Upper-body",
        n_samples=1,
        n_steps=20,
        image_scale=2.0,
        seed=-1,
        api_name="/process_dc"
    )
    print("Result DC:", result_dc)
except Exception as e:
    print("Error in /process_dc:", e)

